src/components/TextInput/TextInput.js

```diff
@@ -1,102 +1,144 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
+import PasswordInput from './PasswordInput';
+import ControlledPasswordInput from './ControlledPasswordInput';
+import { textInputProps } from './util';
 
-const TextInput = ({
-  labelText,
-  className,
-  id,
-  placeholder,
-  type,
-  onChange,
-  onClick,
-  hideLabel,
-  invalid,
-  invalidText,
-  helperText,
-  light,
-  ...other
-}) => {
-  const textInputProps = {
-    id,
-    onChange: evt => {
-      if (!other.disabled) {
-        onChange(evt);
-      }
-    },
-    onClick: evt => {
-      if (!other.disabled) {
-        onClick(evt);
-      }
-    },
-    placeholder,
-    type,
-  };
+export default class TextInput extends React.Component {
+  static PasswordInput = PasswordInput;
```

Comment:
Can we change `<TextInput>` back to functional given this is still stateless? For static properties here, I see a couple of choices:

1. `TextInput.PasswordInput`
2. Named exports (in [`index.js`](https://github.com/IBM/carbon-components-react/blob/v6.46.0/src/components/TextInput/index.js))

The latter is needed even if we take the former.